core/services/ocr2/plugins/ocr2keeper/log_provider.go

```diff
@@ -56,15 +81,62 @@ func NewLogProvider(
 	}
 
 	return &LogProvider{
-		logger:          logger,
-		logPoller:       logPoller,
-		FilterName:      filterName,
-		registryAddress: registryAddress,
-		lookbackBlocks:  lookbackBlocks,
-		registry:        contract,
+		logger:            logger,
+		logPoller:         logPoller,
+		FilterName:        filterName,
+		registryAddress:   registryAddress,
+		lookbackBlocks:    lookbackBlocks,
+		registry:          contract,
+		client:            client,
+		packer:            pluginevm.NewEvmRegistryPackerV2_0(abi),
+		txCheckBlockCache: pluginutils.NewCache[string](time.Hour),
+		cacheCleaner:      pluginutils.NewIntervalCacheCleaner[string](time.Minute),
 	}, nil
 }
 
+func (c *LogProvider) Start(ctx context.Context) error {
```

Comment:
Start/Close/Ready/Healthy functions are based upon similar pattern in registry implementation